serde_derive/src/internals/ast.rs

```diff
@@ -36,6 +36,42 @@ pub struct Variant<'a> {
     pub original: &'a syn::Variant,
 }
 
+/// An enum representing the distribution of different variant names
+#[derive(PartialEq, Eq, Clone, Copy, Debug)]
+pub enum VariantMix {
+    OnlyStrings,
+    OnlyIntegers,
+    OnlyBooleans,
+    Any,
+}
+
+impl VariantMix {
+    pub fn from_single(variant: &Variant) -> Self {
+        match variant.attrs.name().serialize_name() {
```

Comment:
This is using `serialize_name()`. That is not the right name to pay attention to during generating a Deserialize impl.

`#[serde(rename(serialize = 1, deserialize = false))]`